ros2service/ros2service/api/__init__.py

```diff
@@ -84,3 +88,33 @@ def __init__(self, *, service_type_key=None):
     def __call__(self, prefix, parsed_args, **kwargs):
         service = get_service(getattr(parsed_args, self.service_type_key))
         return [message_to_yaml(service.Request())]
+
+
+def profile_configure_short_keys(
+    profile: rclpy.qos.QoSProfile = None, reliability: str = None,
+    durability: str = None, depth: Optional[int] = None, history: str = None,
+) -> rclpy.qos.QoSProfile:
+    """Configure a QoSProfile given a profile, and optional overrides."""
+    if history:
+        profile.history = rclpy.qos.QoSHistoryPolicy.get_from_short_key(history)
+    if durability:
+        profile.durability = rclpy.qos.QoSDurabilityPolicy.get_from_short_key(durability)
+    if reliability:
+        profile.reliability = rclpy.qos.QoSReliabilityPolicy.get_from_short_key(reliability)
+    if depth and depth >= 0:
+        profile.depth = depth
+    else:
+        if (profile.durability == rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL
+                and profile.depth == 0):
+            profile.depth = 1
+
+
+def qos_profile_from_short_keys(
+    preset_profile: str, reliability: str = None, durability: str = None,
+    depth: Optional[int] = None, history: str = None,
+) -> rclpy.qos.QoSProfile:
+    """Construct a QoSProfile given the name of a preset, and optional overrides."""
+    # Build a QoS profile based on user-supplied arguments
+    profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(preset_profile)
+    profile_configure_short_keys(profile, reliability, durability, depth, history)
+    return profile
```

Comment:
there methods are almost duplication of `ros2topic.api`, can we reuse the already existed methods? probably move those methods to `ros2node.api` the called from else where? if more appropriate location to put, that would be better but i am not really sure...

Reply:
I will have a look into this. Thanks